Documentation/allow_legacy_mode.md

```diff
@@ -0,0 +1,101 @@
+## Allow-legacy mode
+
+etcd v2.0 uses internal API v2 protocol, which is not compatible with the v1 protocol.
+
+Allow-legacy mode is introduced to help user upgrade v0.4 etcd cluster easily, and ensures cluster has a short unavailable period, which is less than 1 minute in most cases (based on the size of the actual data).
+
+### Setup
+
+This mode is on if `ETCD_ALLOW_LEGACY_MODE` is set to true, or etcd is running in CoreOS system.
+
+It treats `ETCD_BINARY_DIR` as the directory for etcd binaries, which is organized in this way:
+
+```
+ETCD_BINARY_DIR
+  |
+  -- 1
+  |
+  -- 2
+```
+
+`1` is etcd with internal API v1 protocol. You should use etcd v0.4.7 here. `2` is etcd with internal API v2 protocol, which is etcd v2.0.
+
+The default value for `ETCD_BINARY_DIR` is `/usr/libexec/etcd/internal_versions/`.
+
+### Usage
+
+When bootstrapping with v1 data directory and v1 flags, etcd executes v0.4.7 binary and runs exactly the same as before.
+
+Before upgrade, you SHOULD check the health of the cluster by running:
+
+```
+$ etcdctl cluster-health
+```
+
+If the cluster and all members are healthy, you can start the upgrading process. Or you should check the unhealthy machines and fix them.
```

Comment:
g/Or.*/d
"If not, check the unhealthy machines and repair them \[using this guide\](link to trobleshooting guide)"